apps/user-portal/src/components/account-recovery-code.tsx

```diff
@@ -0,0 +1,230 @@
+import { getProfileInfo, updateProfileInfo } from "../actions";
+import { MFAIcons } from "../configs";
+import { NotificationActionPayload } from "../models/notifications";
+import { EditSection } from "./edit-section";
+import { ThemeIcon } from "./icon";
+
+/**
+ * Proptypes for the SMS OTP component.
+ */
+interface CodeRecoveryProps {
+    onNotificationFired: (notification: NotificationActionPayload) => void;
+}
+
+/**
+ * SMS OTP section.
+ *
+ * @return {JSX.Element}
+ */
+export const CodeRecovery: React.FunctionComponent<CodeRecoveryProps> = (
+    props: CodeRecoveryProps
+): JSX.Element => {
+    const [mobile, setMobile] = useState("");
+    const [editedMobile, setEditedMobile] = useState("");
+    const [isEdit, setIsEdit] = useState(false);
+    const {t} = useTranslation();
+    const {onNotificationFired} = props;
+
+    const handleUpdate = () => {
+        const data = {
+            Operations: [
+                {
+                    op: "replace",
+                    value: {}
+                }
+            ],
+            schemas: [
+                "urn:ietf:params:scim:api:messages:2.0:PatchOp"
+            ]
+        };
+
+        const phoneNumbers = {
+            type: "mobile",
+            value: mobile
+        };
+
+        data.Operations[0].value = {
+            phoneNumbers: [
+                {
+                    type: "mobile",
+                    value: editedMobile
+                }
+            ]
+        };
+
+        updateProfileInfo(data)
+            .then((response) => {
+                if (response.status === 200) {
+                    onNotificationFired({
+                        description: t(
+                            "views:securityPage.multiFactor.smsOtp.notification.success.description"
+                        ),
+                        message: t(
+                            "views:securityPage.multiFactor.smsOtp.notification.success.message"
+                        ),
+                        otherProps: {
+                            positive: true
+                        },
+                        visible: true
+                    });
+                    getProfileInfo()
+                        .then((res) => {
+                            setMobileNo(res);
+                        });
+                    setIsEdit(false);
+                } else {
+                    onNotificationFired({
+                        description: t(
+                            "views:userProfile.notification.getProfileInfo.error.description"
+                        ),
+                        message: t(
+                            "views:userProfile.notification.getProfileInfo.error.message"
+                        ),
+                        otherProps: {
+                            negative: true
+                        },
+                        visible: true
+                    });
+                }
+            });
+    };
```

Comment:
Why are we updating the phone number here?

Reply:
This section has been removed.